src/Gateways/AbstractGateway.php

```diff
@@ -238,7 +238,8 @@ public function makeRegularPayment(array $order, CreditCardInterface $creditCard
             $order,
             PosInterface::MODEL_NON_SECURE
         );
-        $this->eventDispatcher->dispatch($event);
+        /** @var RequestDataPreparedEvent $event */
+        $event = $this->eventDispatcher->dispatch($event);
```

Comment:
these changes introduce a breaking change in laravel-pos library, because of the custom event dispatcher we implemented which does not return $event object.